lib/charms/layer/openstack.py

```diff
@@ -138,9 +138,10 @@ def detect_octavia():
         creds = _load_creds()
         region = creds['region']
         for catalog in _openstack('catalog', 'list'):
-            if (catalog['Name'] == 'octavia' and catalog.get('Endpoints')
-                    and catalog['Endpoints'][0]['region'] == region):
-                return True
+            if catalog['Name'] == 'octavia':
+                for endpoint in catalog.get('Endpoints', []):
```

Comment:
It does not make a lot of sense to default the get value to [], as you wont iterate over the result. Also you should handle the case where the default ([]) would be returned.
for endpoint in catalog.get('Endpoints', None):
```python
if endpoint is not None and endpoint['region'] == region:
...
```

Reply:
@sombrafam thanks for the review.
I don't see a problem in returning empty list when endpoints are not defined. It is expected to skip the for loop and detect_octavia() returns False at #L149 in those cases.
Moreover I see the recommended code snippet requires extra computation for every endpoint to check if it is None or not which is unnecessary.

Let me know if you think otherwise.